src/pages/Account/Venue/VenueMapEditor/Container.tsx

```diff
@@ -0,0 +1,242 @@
+          (imageDim * safeZoneDimension))
+      );
+    };
+    //need to return the unscaled values
+    const unscaledBoxes = boxes.map(
+      (box) =>
+        ({
+          ...box,
+          width: (100 * box.width) / (imageDims.width * safeZonePerc.width),
+          height: (100 * box.height) / (imageDims.height * safeZonePerc.height),
+          top: convertDisplayedCoordToIntrinsic(
+            box.top,
+            "height",
+            safeZonePerc.height
+          ),
+          left: convertDisplayedCoordToIntrinsic(
+            box.left,
+            "width",
+            safeZonePerc.width
+          ),
+        } as SubVenueIcon)
+    );
+
+    onChange && onChange(unscaledBoxes);
+  }, [boxes, onChange, imageDims, safeZonePerc.width, safeZonePerc.height]);
+
+  useMemo(() => {
+    if (!imageDims || !imageDims.width || !imageDims.height) return;
+
+    const copy = iconsMapPerc.map(
+      (mapIcon) =>
+        ({
+          ...mapIcon,
+          width: imageDims.width * mapIcon.width * safeZonePerc.width,
+          height: imageDims.height * mapIcon.height * safeZonePerc.height,
+          top:
+            imageDims.height * ((1 - safeZonePerc.height) / 2) +
+            imageDims.height * mapIcon.top * safeZonePerc.height,
+          left:
+            imageDims.width * ((1 - safeZonePerc.width) / 2) +
+            imageDims.width * mapIcon.left * safeZonePerc.width,
+        } as SubVenueIcon)
+    );
+
+    setBoxes(copy);
+  }, [iconsMapPerc, imageDims, safeZonePerc]);
+
+  const moveBox = useCallback(
+    (id: number, left: number, top: number) => {
+      setBoxes(
+        update(boxes, {
+          [id]: {
+            $merge: { left, top },
+          },
+        })
+      );
+    },
+    [boxes]
+  );
+  const resizeBox = useCallback(
+    (id: number) => (dimensions: Dimensions) => {
+      const { width, height } = dimensions;
+      setBoxes(
+        update(boxes, {
+          [id]: {
+            $merge: { width, height },
+          },
+        })
+      );
+    },
+    [boxes]
+  );
+
+  const [, drop] = useDrop({
+    accept: ItemTypes.SUBVENUE_ICON,
+    drop: (item: DragItem, monitor) => {
+      if (!interactive) return;
+      const delta = monitor.getDifferenceFromInitialOffset() as {
+        x: number;
+        y: number;
+      };
+
+      let left = Math.round(item.left + delta.x);
+      let top = Math.round(item.top + delta.y);
+      if (snapToGrid) {
+        [left, top] = doSnapToGrid(left, top);
+      }
+
+      moveBox(item.itemId, left, top);
+    },
+  });
+
+  return (
+    <>
+      <div ref={drop}>
+        <div ref={ref} className={styles.mapContainer}>
+          <ReactResizeDetector handleWidth handleHeight>
+            {({ targetRef }) => <span ref={targetRef} />}
+          </ReactResizeDetector>
+          <img
+            alt="draggable background"
+            style={{
+              width: "100%",
+            }}
```

Comment:
Inline style. 

[Q] Would a class be more useful here?
That being asked, it is most likely OOS to determine if this container is admin or attendee side and deal accordingly